packages/shared/lib/contexts/onboarding/helpers/convertOnboardingProfileToPersistedProfile.ts

```diff
@@ -1,13 +1,14 @@
-import { IPersistedProfile } from '@core/profile'
-import { DEFAULT_ACTIVE_PROFILE_VALUE } from '@core/profile/constants/default-active-profile-values.constant'
+import { IPersistedProfile } from '@core/profile/interfaces'
+import { DEFAULT_PERSISTED_PROFILE_OBJECT } from '@core/profile/constants'
 
 import { IOnboardingProfile } from '../interfaces'
+import { sanitizeOnboardingProfile } from './sanitizeOnboardingProfile'
 
 export function convertOnboardingProfileToPersistedProfile(
     onboardingProfile: Partial<IOnboardingProfile>
 ): IPersistedProfile {
     return {
-        ...DEFAULT_ACTIVE_PROFILE_VALUE,
-        ...onboardingProfile,
+        ...DEFAULT_PERSISTED_PROFILE_OBJECT,
+        ...sanitizeOnboardingProfile(onboardingProfile),
```

Comment:
if we pass in here an `IOnboardingProfile` and only sanitize `importFile`, `importFilePath`, `mnemonic` and `strongholdPassword`, the resulting object will contain more fields than the return type of `convertOnboardingProfileToPersistedProfile`, e.g. `protectionType`, ` hasStoreMnemonic` etc. 

Either we need to update the return type or make sure that only the relevant fields are returned

Reply:
Updated